fdbclient/include/fdbclient/BlobRestoreCommon.h

```diff
@@ -35,4 +35,44 @@ struct BlobGranuleBackupConfig : public KeyBackedClass {
 	KeyBackedProperty<std::string> mutationLogsUrl() { return subspace.pack(__FUNCTION__sr); }
 };
 
-#endif
\ No newline at end of file
+// Defines blob restore state
+enum BlobRestorePhase {
+	UNINIT = 0,
+	INIT = 1,
+	STARTING_MIGRATOR = 2,
+	LOADING_MANIFEST = 3,
+	LOADED_MANIFEST = 4,
+	COPYING_DATA = 5,
+	APPLYING_MLOGS = 6,
+	DONE = 7,
+	ERROR = 8,
+	MAX = 9
+};
+
+template <>
+inline Standalone<StringRef> TupleCodec<BlobRestorePhase>::pack(BlobRestorePhase const& val) {
+	return Tuple::makeTuple(static_cast<int>(val)).pack();
+}
+template <>
+inline BlobRestorePhase TupleCodec<BlobRestorePhase>::unpack(Standalone<StringRef> const& val) {
+	return static_cast<BlobRestorePhase>(Tuple::unpack(val).getInt(0));
+}
+
+struct BlobGranuleRestoreConfig : public KeyBackedClass {
+	BlobGranuleRestoreConfig(KeyRef prefix = SystemKey("\xff\x02/bgrestore/"_sr)) : KeyBackedClass(prefix) {}
+
+	KeyBackedProperty<UID> uid() { return subspace.pack(__FUNCTION__sr); }
```

Comment:
It occurs to me that making a local static value won't work in this case if you allow the prefix to be configurable, since you need different key-backed instances for each prefix. There may be other ways you could structure this if it matters, but this is also similar to how other of these types are being used in backup and so it may not be a big deal here.

Reply:
thanks! Just follows backup config and leave as is